modules/calib3d/test/test_solvepnp_ransac.cpp

```diff
@@ -2258,4 +2259,51 @@ TEST(Calib3d_SolvePnP, inputShape)
     }
 }
 
+bool hasNan(const cv::Mat& mat)
+{
+    bool has = false;
+    if (mat.type() == CV_32F)
+    {
+        for(size_t i = 0; i < mat.total(); i++)
+            has |= cvIsNaN(mat.at<float>(i));
+    }
+    else if (mat.type() == CV_64F)
+    {
+        for(size_t i = 0; i < mat.total(); i++)
+            has |= cvIsNaN(mat.at<double>(i));
+    }
+    else
+    {
+        has = true;
+        CV_LOG_ERROR(NULL, "check hasNan called with unsupported type!");
+    }
+
+    return has;
+}
```

Comment:
Can this check be replaced to `countNonZero(mat != mat) != 0`?

Reply:
No, this trick does not work with OpenCV.
https://stackoverflow.com/questions/41759247/filter-opencv-mat-for-nan-values#